app/views/directives/create-secret.html

```diff
@@ -0,0 +1,307 @@
+<ng-form name="secretForm">
+  <div for="secretType" ng-if="!type" class="form-group">
+    <label>Secret Type</label>
+    <ui-select required ng-model="newSecret.type" search-enabled="false" ng-change="newSecret.authType = secretAuthTypeMap[newSecret.type].authTypes[0]">
+      <ui-select-match>{{$select.selected + ' Secret'}}</ui-select-match>
+      <ui-select-choices repeat="type in secretTypes">
+        {{type  + ' Secret'}}
+      </ui-select-choices>
+    </ui-select>
+  </div>
+
+  <div class="form-group">
+    <label for="secretName" class="required">Secret Name</label>
+      <div>
+        <input class="form-control"
+          id="secretName"
+          name="secretName"
+          ng-model="newSecret.data.secretName"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
+      </div>
+      <div class="help-block">
+        Unique name of the new secret.
+      </div>
+  </div>
+  <div class="form-group">
+    <label for="authentificationType">Authetication Type</label>
+    <ui-select required ng-model="newSecret.authType" search-enabled="false">
+      <ui-select-match>{{$select.selected}}</ui-select-match>
+      <ui-select-choices repeat="type in secretAuthTypeMap[newSecret.type].authTypes">
+        {{type}}
+      </ui-select-choices>
+    </ui-select>
+  </div>
+
+  <div ng-if="newSecret.authType === 'Basic Authentication'">
+    <div class="form-group">
+      <label for="username">Username</label>
+      <div>
+        <input class="form-control"
+          id="username"
+          name="username"
+          ng-model="newSecret.data.username"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false">
+      </div>
+      <div class="help-block">
+        Optional username for SCM servers authentication.
+      </div>
+    </div>
+
+    <div class="form-group">
+      <label for="passwordToken" class="required">Password or Token</label>
+      <div>
+        <input class="form-control"
+          id="passwordToken"
+          name="passwordToken"
+          ng-model="newSecret.data.passwordToken"
+          type="text"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          required>
+      </div>
+      <div class="help-block">
+        Password or token for SCM servers authentication.
+      </div>
+    </div>
+  </div>
+
+  <div ng-if="newSecret.authType === 'SSH Key'">
+    <div class="form-group" id="private-key">
+      <label for="privateKey" class="required">SSH Private Key</label>
+      <osc-file-input 
+        id="privateKey"
+        model="newSecret.data.privateKey"
+        drop-zone-id="private-key"
```

Comment:
Is it a mistake that one ID is `privateKey` and the other is `private-key`? If they are meant to be different, I wouldn't give them names that are so close.